rosidl_runtime_rs/src/string.rs

```diff
@@ -224,10 +226,47 @@ macro_rules! string_impl {
             }
         }
 
+        impl FromIterator<char> for $string {
+            fn from_iter<I: IntoIterator<Item = char>>(iter: I) -> Self {
+                let mut buf = <$string>::default();
+                buf.extend(iter);
+                buf
+            }
+        }
+
+        impl<'a> FromIterator<&'a char> for $string {
+            fn from_iter<I: IntoIterator<Item = &'a char>>(iter: I) -> Self {
+                let mut buf = <$string>::default();
+                buf.extend(iter);
+                buf
+            }
+        }
+
+        impl Extend<char> for $string {
+            fn extend<I: IntoIterator<Item = char>>(&mut self, iter: I) {
+                let mut v = self.to_vec();
+                let iterator = iter.into_iter();
+
+                iterator.for_each(|c| {
+                    v.push(c as $char_type);
+                });
+
+                // SAFETY: It's okay to pass a non-zero-terminated string here since assignn
+                // uses the specified length and will append the 0 to the dest string itself.
+                if !unsafe { $assignn(self as *mut _, v.as_ptr() as *const _, v.len()) } {
+                    panic!("$assignn failed");
+                }
+            }
```

Comment:
The strings are UTF-8/UTF-16, but this implementation only works for ASCII strings – you can't cast a `char` to `$char_type`. One way to do it properly would be to get a `std::string::String` from `$string`, extend that, and convert it back to the `$string` type. It might be possible to still do better than that (performance-wise), but that would be a good start.

Reply:
Oops, good call. Fixed